### ISSUE TITLE
False positives when providing default values in destructuring assignment

### ISSUE THREAD

**Issue by lekkas** [za-creature/coffeescope#17]
Hey,

First of all, thanks a lot for sharing this project, it enables are some very useful features missing from `coffeelint` 👍 

I've bumped into an edge case when combining destructuring with default values. Here is the snippet in question:

``` coffeescript
defaultVal = 0 
{ property = defaultVal } = {}
property
```

Also the `coffeelint.json` I used:

``` json
{
    "check_scope": {
      "module": "coffeescope2",
      "undefined": true,
      "unused_variables": true
    }
}
```

Which reports:

```
     ⚡ #1: Variable "defaultVal" is never used.
     ⚡ #2: Variable "defaultVal" is never used (first defined on line 1).
     ⚡ #3: Undefined identifier "property".
```

In this case, `defaultVal` is actually used (as a default parameter) and `property` isn't undefined after the destructuring assignment.

**Comment by za-creature** [comment]
Thanks for the report; released in 0.3.3

**Comment by lekkas** [comment]
That was blazing fast - thank you!
